activation/nipost_test.go

```diff
@@ -748,11 +761,12 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 			zaptest.NewLogger(t),
 			poetCfg,
 			mclock,
+			nil,
 			WithPoetClients(poetProver),
 		)
 		require.NoError(t, err)
 
-		nipst, err := nb.BuildNIPost(context.Background(), sig, postGenesisEpoch+2, challenge)
+		nipst, err := nb.BuildNIPost(context.Background(), sig, postGenesisEpoch+2, challenge, nil)
```

Comment:
NIT: the new `postChallenge` parameter should never be `nil`. Only its `InitialPost` field will be `nil` in most cases.

Maybe it makes sense to change the method signature to:

```go
BuildNIPost(
	ctx context.Context,
	sig *signing.EdSigner,
	challengeHash types.Hash32,
	postChallenge *types.NIPostChallenge,
) (*nipost.NIPostState, error)
```

and drop the `publish` epoch parameter since it is taken from the challenge anyway. Then this line would become

```suggestion
		nipst, err := nb.BuildNIPost(context.Background(), sig, challenge, &types.NIPostChallenge{PublishEpoch: postGenesisEpoch+2})
```

Wdyt?

Reply:
done